src/shader/shaderLightSampling.js

```diff
@@ -1,11 +1,51 @@
 export const shaderLightSampling = /* glsl */`
 
+float getSpotAttenuation( const in float coneCosine, const in float penumbraCosine, const in float angleCosine ) {
+
+	return smoothstep( coneCosine, penumbraCosine, angleCosine );
+
+}
+
+float getDistanceAttenuation( const in float lightDistance, const in float cutoffDistance, const in float decayExponent ) {
+
+		// based upon Frostbite 3 Moving to Physically-based Rendering
+		// page 32, equation 26: E[window1]
+		// https://seblagarde.files.wordpress.com/2015/07/course_notes_moving_frostbite_to_pbr_v32.pdf
+		float distanceFalloff = 1.0 / max( pow( lightDistance, decayExponent ), EPSILON );
+
+		if ( cutoffDistance > 0.0 ) {
+			distanceFalloff *= pow2( saturate( 1.0 - pow4( lightDistance / cutoffDistance ) ) );
+		}
+
+		return distanceFalloff;
+}
+
+// float atan2(float y, float x) {
+
+//     bool s = ( abs( x ) > abs( y ) );
+//     return mix( PI / 2.0 - atan( x, y ), atan( y, x ), s );
+
+// }
```

Comment:
Is this commented code still needed?

Reply:
It is if we want to use the other dimension IESLoader provides, most profiles don't end up using it but one (I forget which) from the set in the `iesProfiles` array does. 

The profiles are 360x180 and repeat the rows anyway, so we can use this and just pay the extra cost. It's a case of deciding whether it's worth it, if we decide to stick with 1D profiles then I don't need to use a texture array.